chainstate/src/detail/chainstateref/mod.rs

```diff
@@ -790,27 +792,57 @@ impl<'a, S: BlockchainStorageWrite, O: OrphanBlocksMut, V: TransactionVerificati
             )
             .log_err()?;
 
-        let consumed = connected_txs.consume()?;
-        flush_to_storage(self, consumed)?;
+        let consumed = tx_verifier.consume().log_err()?;
+        self.flush_to_storage_after_connect(block_index, consumed)
+    }
 
-        Ok(())
+    fn flush_to_storage_after_connect(
+        &mut self,
+        block_index: &BlockIndex,
+        consumed_verifier_data: TransactionVerifierDelta,
+    ) -> Result<(), BlockError> {
+        // Flush accounting delta
+        self.db_tx
+            .set_accounting_delta(
+                *block_index.block_id(),
+                &consumed_verifier_data.accounting_delta().clone(),
```

Comment:
Exposing accounting data from `TransactionVerifierDelta` seems like a hacky way of storing delta per block. But it greatly simplifies things. Just one additional function call.

Another more encapsulated way of doing it is to teach `TransactionVeirifer` to flush delta in `flush_to_storage` function. But the problem is that TransactionVerifier can accumulate info from several blocks, which requires additional logic and fields that would store deltas per block, accumulate them, reverse on disconnect, and so on. Considering that these deltas per block don't relate to the verification of transactions in any way, it seems to me like a bigger evil than consumed data exposure.

Any thoughts?